theano/gpuarray/basic_ops.py

```diff
@@ -973,7 +973,22 @@ def empty_like(var):
     return GpuAllocEmpty(var.type.dtype, var.type.context_name)(*var.shape)
 
 
-def gpu_alloc_empty(dtype, ctx):
+def gpu_alloc_empty(ctx, **kwargs):
+    '''
+    This is the cache method of GpuAllocEmpty class.
+    This takes the parameters of context name and props_dict
+    and retrieves the dtype key from the dictionary
+
+    Parameters
+    ----------
+    ctx : String
+        The context name.
+
+    kwargs : Dict
+        The props_dict of the Op
+
+    '''
+    dtype = kwargs.get('dtype')
```

Comment:
What? No, you don't need to change the order of the arguments, that is just more confusing.
Moreover, you just end up ignoring all arguments that are not 'dtype', which defeats the purpose of making sure the optimization crashes if the CPU op has new, unknown arguments.